tofhir-engine/src/main/scala/io/tofhir/engine/model/FhirMappingJobExecution.scala

```diff
@@ -137,5 +136,44 @@ case class FhirMappingJobExecution(id: String = UUID.randomUUID().toString,
     // Combine the transformed words to create a folder name
     extractedWords.mkString("-")
   }
+}
 
+/**
+ * An object to create FhirMappingJobExecution instances
+ */
+object FhirMappingJobExecution {
+  def apply(id: String = UUID.randomUUID().toString,
+            projectId: String = "",
+            job: FhirMappingJob,
+            mappingTasks: Seq[FhirMappingTask] = Seq.empty,
+            jobGroupIdOrStreamingQuery: Option[Either[String, collection.mutable.Map[String, StreamingQuery]]] = None
+           ): FhirMappingJobExecution = {
+    val jobId = if (job != null && job.id != null) job.id else ""
+
+    val sourceSettings = Option(job.sourceSettings)
```

Comment:
Do we really need this Option ? Source settings are mandatory in the fhir mapping job model.

Reply:
No, we dont need Option. I am confused about the empty Map.